module-2/backend/week_two.md

```diff
@@ -26,20 +30,57 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+`team = Team.find(4) team.owner`
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+A student has many teachers and a teacher has many students.
+This is known as a many to many relationship. We would have to specify a joins table to accurately model this relationship. Maybe with a teacher_students table?
 6. Define foreign key, primary key, and schema.
+A foreign key is how we can access another tables primary key. It acts as the link between two tables. A primary key is a resources primary identifier, usually an ID. A schema is the current 'blueprint' for the database. it defines the tables and the rows and columns inside those tables as well as the relationships between said tables.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+A foreign key on one table is the primary key on a seperate table.
 8. What are the parts of an HTTP response?
-
+status, headers, body
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+
+`find_or_create` is a wonderful method that cancels out the need to do an if else blah blah blah when creating a new resource.
+Want an array of column values for certain records? => => `pluck`
+
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+
+`rake db:drop db:create db:migrate` "I messed everything up horribly. please help"
+
 3. What two columns does `t.timestamps null: false` create in our database?
+
+two timestamps => created at and updated at.
+
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+
+A school has many teachers and a teacher belongs to a school
+
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+
+School  => has_many   :teachers
+
+Teacher => belongs_to :school
+
 6. Give an example of when you might want to store information besides ids on a join table.
+
+We joined on date for bike share. That was interesting.
```

Comment:
Ha! I was thinking of something like `invoice_items`, which stores a quantity and unit price. Or a `courses` table (join between students and teachers) that could store a grade for that course and maybe the semester that they took the course.